witness/golang/omniwitness/usbarmory/storage/slots/journal.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package slots provides a simple "postbox" type filesystem.
+package slots
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// magic0 is the only known journal header prefix.
+const magic0 = "TFJ0"
+
+// BlockReaderWriter describes a type which knows how to read and write
+// whole blocks to some backing storage.
+type BlockReaderWriter interface {
+	// BlockSize returns the block size of the underlying storage system.
+	BlockSize() uint
+
+	// ReadBlocks reads len(b) bytes into b from contiguous storage blocks starting
+	// at the given block address.
+	// b must be an integer multiple of the device's block size.
+	ReadBlocks(lba uint, b []byte) error
+
+	// WriteBlocks writes len(b) bytes from b to contiguous storage blocks starting
+	// at the given block address.
+	// b must be an integer multiple of the device's block size.
+	WriteBlocks(lba uint, b []byte) error
+}
+
+// Journal implements a record-based format which provides a resilient storage.
+type Journal struct {
+	dev           BlockReaderWriter
+	start         uint
+	length        uint
+	current       entry
+	nextBlock     uint
+	maxWriteBytes uint
```

Comment:
`maxDataBytes` is a little more precise (if the `Update` method was called `Write` I would have let this slide).

Reply:
Good spot, ta